plugins/vst_base/VstPlugin.cpp

```diff
@@ -128,23 +132,49 @@ VstPlugin::VstPlugin( const QString & _plugin ) :
 {
 	setSplittedChannels( true );
 
-	PE::MachineType machineType;
-	try {
-		PE::FileInfo peInfo(m_plugin);
-		machineType = peInfo.machineType();
-	} catch (std::runtime_error& e) {
-		qCritical() << "Error while determining PE file's machine type: " << e.what();
-		machineType = PE::MachineType::unknown;
+	auto pluginType = ExecutableType::Unknown;
+#ifdef LMMS_BUILD_LINUX
+	QFileInfo fi(m_plugin);
+	if (fi.suffix() == "so")
+	{
+		pluginType = ExecutableType::Linux64;
+	}
+	else
+#endif
+	{
+		try {
+			PE::FileInfo peInfo(m_plugin);
+			switch (peInfo.machineType())
+			{
+			case PE::MachineType::amd64:
+				pluginType = ExecutableType::Win64;
+				break;
+			case PE::MachineType::i386:
+				pluginType = ExecutableType::Win32;
+				break;
+			default:
+				qWarning() << "Unknown PE machine type"
+					<< QString::number(static_cast<uint16_t>(peInfo.machineType()), 16);
+				break;
```

Comment:
Not sure about others, but I prefer wrapping case blocks in `{}`

Reply:
I personally prefer to leave them, unless the case block contains a variable declaration, in which case braces are often necessary. Shall we make this part of the clang-format discussion, and let the auto-formatter tidy it up later if we decide to use braces?

Reply:
That sounds good to me.